roofit/xroofit/src/xRooFit.cxx

```diff
@@ -1124,9 +1117,12 @@ int xRooFit::minos(RooAbsReal &nll, const RooFitResult &ufit, const char *parNam
    fitConfig.SetMinosErrors(mErrs);
    par->setConstant(isConst);
 
-   const_cast<RooFitResult &>(ufit)._statusHistory.push_back(
-      std::make_pair(std::string(TString::Format("xMINOS_%s", parName)), status));
-   const_cast<RooFitResult &>(ufit)._status += status;
+   std::vector<std::pair<std::string,int>> statusHistory;
+   for(int i=0;i<ufit.numStatusHistory();i++)
+      statusHistory.push_back(std::make_pair(ufit.statusLabelHistory(i),ufit.statusCodeHistory(i)));
+   statusHistory.push_back(std::make_pair(std::string(TString::Format("xMINOS_%s", parName)), status));
```

Comment:
```suggestion
   statusHistory.emplace_back(TString::Format("xMINOS_%s", parName), status);
```
Besides using `emplace_back`, you can even make the construction of the string here implicit